core/src/main/clojure/xtdb/metadata.clj

```diff
@@ -351,6 +354,27 @@
         (when (.getObject bloom-rdr bloom-vec-idx)
           (bloom/bloom->bitmap bloom-rdr bloom-vec-idx)))))
 
+  (temporalBounds[_ page-idx]
+    ;; TODO when badly distributed there are pages with no data
+    (if-let [temporal-col-types-rdr (-> (.structKeyReader metadata-leaf-rdr "columns")
+                                        (.listElementReader)
+                                        (.structKeyReader "types")
+                                        (.structKeyReader temporal-col-type-leg-name))]
+      (let [min-rdr (.structKeyReader temporal-col-types-rdr "min")
+            max-rdr (.structKeyReader temporal-col-types-rdr "max")
+            ^long system-from-idx (.get page-idx-cache (PageIndexKey. "xt$system_from" page-idx))
+            bounds (TemporalBounds.)]
+        ;; assumes sys-from, valid-from, valid-to always appear in that order
```

Comment:
this is what the page-idx-cache is for - unless/until it shows up in the profiling I'd err on the side of being safe